### ISSUE TITLE
Research on travel status verification method

### ISSUE THREAD

**Issue by Rishikant181** [SURAJ-SHARMA27/just_started#1]

**Comment by SURAJ-SHARMA27** [comment]
Can we consider this approach ?

**PNR Verification:**

- How It Works: When a user enters their PNR (Passenger Name Record), the app can check the train number, date of travel, and coach/berth details from the Indian Railways API (IRCTC or any other reliable source).
- Limitations: As you mentioned, someone could enter the PNR from home before the journey starts. To mitigate this, consider combining PNR verification with real-time location tracking.

**Location Verification:**

- How It Works: The app can use GPS to check if the user is currently near the train station or on the train route. You could prompt the user to enable location services, and the app can periodically check if the user's location matches the train's path.
- Implementation: This could be done by matching the user's location with the train's real-time location (e.g., using APIs like RailYatri, RailRadar, or other train tracking services).

**Hybrid Verification Approach:**

- Combine both PNR and GPS location verification to enhance security. If the PNR and location both match the train's real-time status, then the user can be verified.

**Comment by Rishikant181** [comment]
Yeah, this was the approach I was talking about, and it seems good enough for starters. We can use this approach initially, then, if we find something better, we'll change it.

If possible, we may also incorporate the velocity of user using their GPS, which can be used to further confirm the user's presence in the train, and not just some stationary point alone the route.

**Comment by Rishikant181** [comment]
As for the Railways API, I think there one provided by [Indian Railways](https://crisapis.indianrail.gov.in/devportal/home).
